### PR TITLE
IX adapter code refactoring

### PR DESCRIPTION
**Pull request by indexexchange**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

The change includes following:

- adding more test cases to increase IX adapter code coverage
- improving configuration validation
- adding prebid version in IX demand request parameter
- adding addBidResponse() test case to make sure the function does not override adid. IX returns multiple bids from #638 change. Individual bid's adid should be different.

### CODE REVIEW

**Comment by protonate** [comment]
Thanks for the PR, assigned for review.

File: test/spec/adapters/indexExchange_request_spec.js [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+
+		var configuredBids = [
+			IndexUtils.createBidSlot( IndexUtils.DefaultPlacementCodePrefix, slotID, [ IndexUtils.supportedSizes[0] ], { slotSize : slotSizes } )
+		];
+
+		adapter.callBids({ bids: configuredBids });
+
+		assert.isUndefined( adLoader.loadScript.firstCall.args[0], "no request made to AS");
+	});
+});
```
Comment (protonate) [inline test/spec/adapters/indexExchange_request_spec.js]: Many thanks for the additional test coverage.

**Comment by protonate** [comment]
### No ID in response (topic branch)
This shows the topic branch response, you'll see the ID highlighted in the address bar and on the last line of the response (at the bottom) the ID is **null**.


screen shot 2016-10-24 at 3 31 29 pm




### ID in response (master branch)
This ad renders and the ID is contained in the response.

screen shot 2016-10-24 at 3 31 53 pm

**Comment by indexexchange** [comment]
> Please resolve the ad render for this branch and we can merge.

The issue is resolved from the sandbox server; no adapter code change required.

**Comment by protonate** [comment]
Thanks @indexexchange, merged.

**Pull request merged**